src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/Cas2SubmittedApplicationReport.kt

```diff
@@ -22,8 +22,11 @@ interface Cas2SubmittedApplicationReportRepository : JpaRepository<DomainEventEn
         TO_CHAR(
           CAST(events.data -> 'eventDetails' ->> 'submittedAt' AS TIMESTAMP),
           'YYYY-MM-DD"T"HH24:MI:SS'
-        ) AS submittedAt
+        ) AS submittedAt,
+        TO_CHAR(applications.created_at, 'YYYY-MM-DD"T"HH24:MI:SS') AS createdAt
```

Comment:
May be this is not related to this PR and not aware of background, am curious in understanding the reason behind to use domain_events table instead of applications(cas_2_applications) table to get the data we needed?

Reply:
I have a ready-to-go answer to @sudhakar-plr 's question 😆 
[we discussed this here](https://mojdt.slack.com/archives/C05GEPLBRN3/p1707996480266999)

For application submissions we want to see a log over time of how many submissions took place and at what time, regardless of the state of the database when the report is requested, whereas in the case of in-progress apps we do want a snapshot of the the database at that point in time. (There isn't an 'event', a specific point in time, in the case of in progress apps)

The domain events table is the source of truth for activity over time on our service. The service managers need to know how many events of the type CAS2_APPLICATION_SUBMITTED happened over a given period of time, not what the state of the data is at the point we run the report. This decouples it from any changes to our database. E.g. as an extreme example I can run a report to see that a year ago application with ID '1234' was submitted by what user, even if in the mean time we have completely changed the data shape of an application to no longer have a user.

Reply:
here's some more context from Ed with further reasons - TLDR, making the reports as simple as possible 
https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/1284

Reply:
> This field is [renamed](https://github.com/ministryofjustice/hmpps-approved-premises-api/blob/e7d281662613d8a2ddde708c83d710f43aaa7ae6/src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/Cas2UnsubmittedApplicationsReport.kt#L16) to `startedAt` for the unsubmitted applications report. Should `createdAt` be `startedAt` for this submitted applications report too?

@Harriethw just confirming we are ok with this as it stands and calling it `createdAt` is ok? It means this submitted applications report will contain a `createdAt` field and the unsubmitted applications report contains a `startedAt` field - though they both mean the same thing, it will be a difference in naming convention.  Really the equivalent unsubmitted report field should also be `createdAt` - well I think that should be the case...

Reply:
ah yeah `startedAt` would make more sense I would have thought from a user point of view? I don't think there's any reasoning on the technical side one way or the other... as long as Melissa knows what column to point to

Reply:
Yes agreed, at least have them both mistaken rather than one and not the other - especially since Melissa is expecting it to be `startedAt`.